lib/elastic_apm/agent.rb

```diff
@@ -186,7 +188,7 @@ def set_user(user)
     end
 
     def add_filter(key, callback)
-      @transport.filters.add(key, callback)
+      transport.worker.filters.add(key, callback)
```

Comment:
Not really a fan of this. Not thread safe either. A user could theoretically add a new filter while the agent is applying the existing filters somewhere else. Probably not in _real life projects_ but in theory it isn't safe as is.